src/config.rs

```diff
@@ -149,9 +152,19 @@ fn load_or_build(path: &Path) -> Result<HoardConfig, Error> {
             loaded_config.parameter_token = Some(HoardConfig::default_parameter_token());
             is_config_dirty = true;
         }
+        if loaded_config.read_from_current_directory.is_none() {
+            loaded_config.read_from_current_directory = Some(false);
+            is_config_dirty = true;
+        }
         if is_config_dirty {
             save_config(&loaded_config, &hoard_config_path)?;
         }
+
+        let path_buf = Path::new(HOARD_FILE).to_path_buf();
+        if loaded_config.read_from_current_directory.unwrap() && path_buf.exists() {
+            loaded_config.trove_home_path = Some(path_buf);
+        }
```

Comment:
Added here for commands such as `hoard new`.

Reply:
Now that the loaded trove in `HoardConfig` can be a other trove file other than the one in its home path, we could refactor the name form `trove_home_path` to just `trove_path`